e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh

```diff
@@ -175,27 +175,73 @@ appinsights_connstr=$(az monitor app-insights component show \
 az keyvault secret set --vault-name "$kv_name" --name "applicationInsightsKey" --value "$appinsights_key"
 az keyvault secret set --vault-name "$kv_name" --name "applicationInsightsConnectionString" --value "$appinsights_connstr"
 
+# ###########################
+echo "validate_password function."
+# ###########################
+validate_password() {
+    local password="$1"
+    local max_retries="$2"
+    local retry_count="$3"
+
+    if [[ -z "$password" || "$password" == "-" ]]; then
+        ###if there is an hyphen retry
+        ((retry_count++))
+        if [[ $retry_count -ge $max_retries ]]; then
+            echo "Error: Failed to generate a valid password after $max_retries retries." >&2
+            exit 1
+        fi
+        echo "Invalid password as the hyphen is the first character. Retrying... Attempt $retry_count"
```

Comment:
Minor rewording
Original
"Invalid password as the hyphen is the first character." ....
Suggestion
"Invalid password. The first character cannot be a hyphen." ....

Reply:
it would be better in fact